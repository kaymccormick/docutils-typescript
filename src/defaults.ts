/**
 * @uuid 01ca88e9-b4ab-448d-944d-5af49bb62b08
 */
const defaultReaderName = 'standalone';
const defaultParserName = 'restructuredtext';
const defaultEnableExitStatus = true;

export { defaultReaderName, defaultParserName, defaultEnableExitStatus };
