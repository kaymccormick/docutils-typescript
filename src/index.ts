/**
 *  eslint-disable-next-line import/prefer-default-export 
 * @uuid 62a09a7d-88ab-4fd5-bc03-98b19d2cc427
 */
export { parse } from './parse';
export { StringOutput, StringInput } from './io';
export { StandaloneReader } from './readers/standalone';
export { newDocument } from './newDocument';
import * as nodes from './nodes';
import Writer from './Writer';
import Transform from './Transform';
export { Reader } from './Reader';
export { Publisher } from './Publisher'
export { defaults } from '../gen/defaults';
export { pojoTranslate } from './fn/pojoTranslate';
export { htmlTranslate } from './fn/htmlTranslate';
export { RSTParser } from './parsers/restructuredtext';
export { XMLWriter } from './writers/xml'

export { StringOutput, StringInput } from "./io";
export { StandaloneReader } from "./readers/standalone";
export { newDocument } from "./newDocument";
import * as nodes from "./nodes";
import Writer from "./Writer";
import Transform from "./Transform";
export { Reader } from "./Reader";
export { Publisher } from "./Publisher";
export { defaults } from "../gen/defaults";
export { pojoTranslate } from "./fn/pojoTranslate";
export { htmlTranslate } from "./fn/htmlTranslate";
export { RSTParser } from "./parsers/restructuredtext";
export { XMLWriter } from "./writers/xml";

/* eslint-disable-next-line @typescript-eslint/no-unused-vars,no-unused-vars */
const __docformat__ = "reStructuredText";

export const __version__ = "0.14js";
export { nodes, Writer, Transform };