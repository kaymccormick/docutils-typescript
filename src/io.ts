/**
 * @uuid 2b0c8f26-be6e-45a6-951b-96c70f71f55b
 */
import Input from './io/Input';
import Output from './io/Output';
import { ReadInputCallback, LoggerType } from "./types";

/** Direct string input. 
 * @uuid ae2c2dd1-3766-4ae6-a68c-a3a4a0b71cb3
*/
export class StringInput extends Input {
    public constructor(source: string, logger: LoggerType, sourcePath?: string, encoding?: string, errorHandler?: string) {
        super({source, sourcePath, encoding, errorHandler, logger});
        this.sourcePath = '<string>';
    }

    public read(): Promise<any> {
        return Promise.resolve(this.source);
    }
}

/*
 * @uuid 11de15d8-c529-4ddd-830c-6254641c53dd
*/
export class StringOutput extends Output<string> {
    public constructor(
        logger: LoggerType,
        destination?: string,
        destinationPath?: string,
        encoding?: string,
        errorHandler?: string
    ) {
        super(logger, destination, destinationPath, encoding, errorHandler);
        this.defaultDestinationPath = '<string>';

    }

    public write(data: string): string {
        // self.destination = self.encode(data) // fixme encoding
        if (Array.isArray(data)) {
            data = JSON.stringify(data);
        }
        this.destination = data;
        return this.destination;
    }
}

