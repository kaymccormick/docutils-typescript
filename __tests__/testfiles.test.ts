import { Publisher } from '../src/Core';
import { StringInput, StringOutput } from '../src/io';
import * as nodes from '../src/nodes';
import { getDefaultSettings } from "../src/";
import {NodeInterface} from "../src/types";
import { createLogger } from '../src/testUtils';

class Visitor extends nodes.GenericNodeVisitor {
    /* eslint-disable-next-line @typescript-eslint/camelcase,camelcase,@typescript-eslint/no-unused-vars,no-unused-vars */
    default_departure(node: NodeInterface) {
        /**/
    }
 
    /* eslint-disable-next-line @typescript-eslint/camelcase,camelcae */
    default_visit(node: NodeInterface) {
        if (node.attributes && node.attributes.refuri) {
            //                                console.log(node.attrites.refuri);
            if (!/^https?:\/\//.test(node.attributes.refuri)) {
                const msg = `Invalid refuri ${node.attributes.refuri}`;
                const messages = [node.document!.reporter.warning(msg, [], {})];
                node.add(messages);
            }
        }
    }
}

const path = require('path');
const fs = require('fs');

const testFilesRoot = path.join(__dirname, '../testfiles/forms/');

const table = fs.readdirSync(testFilesRoot, { withFileTypes: true })
    .filter((e: any) => !e.isDirectory())
    .map((e: any) => [e.name,
        fs.readFileSync(path.join(testFilesRoot, e.name), 'utf-8')]);

const defaultArgs = {
    readerName: 'standalone',
    parserName: 'restructuredtext',
    usage: '',
    description: '',
    enableExitStatus: true,
    writerName: 'xml',
};

const defaultSettings = { ...getDefaultSettings() };

test.each(table)('%s', (file, input) => {
    const logger = createLogger();
    const myOpts: any = {};

    const settings = { ...defaultSettings };
    const args = { ...defaultArgs };

    const { readerName, parserName, writerName } = args;
    const source = new StringInput( input, logger);
    const destination = new StringOutput(logger);
    const pub = new Publisher({
        source, destination, settings, logger,
    });
    pub.setComponents(readerName, parserName, writerName);
    return pub.publish({}).then(() => {
        const document = pub.document;
 	const visitor = new Visitor(document!);
 	document!.walkabout(visitor);
        expect(destination.destination!).toMatchSnapshot();
    });

});
