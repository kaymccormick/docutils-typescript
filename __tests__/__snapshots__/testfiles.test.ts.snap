// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`anonymous_reference.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<target ids=\\"1\\" refuri=\\"http://www.python.org\\" anonymous=\\"1\\"></target></document>"
`;

exports[`anonymous_via_two_dots.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<target ids=\\"1\\" refuri=\\"http://www.python.org\\" anonymous=\\"1\\"></target></document>"
`;

exports[`anonymous_via_twodots_no_blankfinish.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<target ids=\\"1\\" refuri=\\"http://www.python.org\\" anonymous=\\"1\\"></target></document>"
`;

exports[`block_quote.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<block_quote>
<paragraph>Block quotes consist of indented body elements:</paragraph><block_quote><paragraph>This theory, that is mine, is mine.</paragraph></block_quote><attribution>Anne Elk (Miss)</attribution></block_quote></document>"
`;

exports[`bullet_from_spec.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<bullet_list bullet=\\"-\\">
<list_item>
<paragraph>This is a bullet list.</paragraph></list_item><list_item><paragraph>Bullets can be &quot;*&quot;, &quot;+&quot;, or &quot;-&quot;.</paragraph></list_item></bullet_list></document>"
`;

exports[`bullet_list__invalid_input.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<bullet_list bullet=\\"*\\">
<list_item>
<paragraph>bullet</paragraph></list_item></bullet_list><system_message line=\\"2\\"><paragraph>Bullet list ends without a blank line; unexpected unindent.</paragraph></system_message><paragraph>invalid</paragraph></document>"
`;

exports[`bullet_list__invalid_input_doctest.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<bullet_list bullet=\\"*\\">
<list_item>
<paragraph>bullet</paragraph></list_item></bullet_list><system_message line=\\"2\\"><paragraph>Bullet list ends without a blank line; unexpected unindent.</paragraph></system_message><doctest_block>&gt;&gt;&gt; foo</doctest_block></document>"
`;

exports[`bullet_list__invalid_input_field_marker.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<bullet_list bullet=\\"*\\">
<list_item>
<paragraph>bullet</paragraph></list_item></bullet_list><system_message line=\\"2\\"><paragraph>Bullet list ends without a blank line; unexpected unindent.</paragraph></system_message><field_list><field><field_name>Hello</field_name><field_body><paragraph>foo</paragraph></field_body></field></field_list></document>"
`;

exports[`bullet_list__invalid_input_line.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<bullet_list bullet=\\"*\\">
<list_item>
<paragraph>bullet</paragraph></list_item></bullet_list><system_message line=\\"2\\"><paragraph>Bullet list ends without a blank line; unexpected unindent.</paragraph></system_message><transition></transition></document>"
`;

exports[`bullet_no_unindent.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<bullet_list bullet=\\"*\\">
<list_item>
<paragraph>bullet</paragraph></list_item></bullet_list></document>"
`;

exports[`citation.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>Hello <citation_reference ids=\\"1\\" refname=\\"goober\\">Goober</citation_reference></paragraph><citation ids=\\"goober\\" names=\\"goober\\"><label>Goober</label><paragraph>Citation.</paragraph></citation></document>"
`;

exports[`comment.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<comment>Comments begin with two dots and a space.  Anything may
follow, except for the syntax of footnotes/citations,
hyperlink targets, directives, or substitution definitions.</comment></document>"
`;

exports[`complex.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<bullet_list bullet=\\"-\\">
<list_item>
<paragraph>This is the first line of a bullet list
item&apos;s paragraph.  All lines must align
relative to the first line.  [1]_</paragraph><block_quote><paragraph>This indented paragraph is interpreted
as a block quote.</paragraph></block_quote></list_item><paragraph>Because it is not sufficiently indented,
this paragraph does not belong to the list
item.</paragraph><footnote ids=\\"1\\" names=\\"1\\"><label>1</label><paragraph>Here&apos;s a footnote.  The second line is aligned
with the beginning of the footnote label.  The &quot;..&quot;
marker is what determines the indentation.</paragraph></footnote></bullet_list></document>"
`;

exports[`definition_list.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<definition_list>
<definition_list_item>
<term>what</term><definition><paragraph>Definition lists associate a term with a definition.</paragraph></definition></definition_list_item><definition_list_item><term>how</term><definition><paragraph>The term is a one-line phrase, and the definition is one
or more paragraphs or body elements, indented relative to
the term.</paragraph></definition></definition_list_item></definition_list></document>"
`;

exports[`definition_list_with_classifier_term.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<definition_list>
<definition_list_item>
<term>term</term><classifier>classifier</classifier><definition><paragraph>test</paragraph></definition></definition_list_item><definition_list_item><term>what</term><definition><paragraph>Definition lists associate a term with a definition.</paragraph></definition></definition_list_item><definition_list_item><term>how</term><definition><paragraph>The term is a one-line phrase, and the definition is one
or more paragraphs or body elements, indented relative to
the term.</paragraph></definition></definition_list_item></definition_list></document>"
`;

exports[`doctest_block.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<block_quote>
<doctest_block>&gt;&gt;&gt; print &apos;Python-specific usage examples; begun with &quot;&gt;&gt;&gt;&quot;&apos;
Python-specific usage examples; begun with &quot;&gt;&gt;&gt;&quot;
&gt;&gt;&gt; print &apos;(cut and pasted from interactive Python sessions)&apos;
(cut and pasted from interactive Python sessions)</doctest_block></block_quote></document>"
`;

exports[`docutils_title.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document ids=\\"docutil-project-documentation-overview\\" names=\\"docutils\\\\ project\\\\ documentation\\\\ overview\\" source=\\"&lt;string&gt;\\" title=\\"Docutils

 Project Documentation Overview\\">
<title><reference name=\\"Docutils\\" refname=\\"docutils\\">Docutils</reference> Project Documentation Overview</title></document>"
`;

exports[`emphasis.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph><emphasis>hello</emphasis></paragraph></document>"
`;

exports[`emphasis_and_inline.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph><emphasis>hello</emphasis> and <strong>goodbye</strong></paragraph></document>"
`;

exports[`emphasis_followed_by_text.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph><emphasis>hello</emphasis> test</paragraph></document>"
`;

exports[`emphasis_preceded_by_text.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>stuff <emphasis>hello</emphasis></paragraph></document>"
`;

exports[`emphasis_surrounded_by_text.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>stuff <emphasis>hello</emphasis> things</paragraph></document>"
`;

exports[`escaping_1.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph><emphasis>escape</emphasis> <literal>with</literal> &quot;&quot;</paragraph></document>"
`;

exports[`escaping_2.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph><emphasis>escape</emphasis> <literal>with</literal> &quot;\\\\&quot;</paragraph></document>"
`;

exports[`excerpt_1.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document ids=\\"configuration-etting\\" names=\\"configuration\\\\ settings:\\" source=\\"&lt;string&gt;\\" title=\\"Configuration settings:\\">
<title>Configuration settings:</title><target ids=\\"footnote\\" names=\\"footnote\\" refuri=\\"../doctree.html#footnote\\"></target><target ids=\\"label\\" names=\\"label\\" refuri=\\"../doctree.html#label\\"></target><target ids=\\"footnote-reference-etting\\" names=\\"footnote_references\\\\ setting\\" refuri=\\"../../uer/config.html#footnote-reference-html4c1-writer\\"></target></document>"
`;

exports[`field_list.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<field_list>
<field>
<field_name>Author</field_name><field_body><paragraph>David Goodger</paragraph></field_body></field><field><field_name>Contact</field_name><field_body><paragraph>docutils-develop@lists.sourceforge.net</paragraph></field_body></field><field><field_name>Revision</field_name><field_body><paragraph>$Revision: 8205 $</paragraph></field_body></field><field><field_name>Date</field_name><field_body><paragraph>$Date: 2017-11-27 03:07:28 -0800 (Mon, 27 Nov 2017) $</paragraph></field_body></field><field><field_name>Copyright</field_name><field_body><paragraph>This document has been placed in the public domain.</paragraph></field_body></field></field_list></document>"
`;

exports[`footnote.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<footnote ids=\\"1\\" names=\\"1\\">
<label>1</label><paragraph>A footnote contains body elements, consistently
indented by at least 3 spaces.</paragraph></footnote></document>"
`;

exports[`footnote_auto-symbol.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>Here is a symbolic footnote reference: <footnote_reference ids=\\"1\\" auto=\\"*\\"></footnote_reference>.</paragraph><footnote ids=\\"2\\" auto=\\"*\\"><paragraph>This is the footnote.</paragraph></footnote></document>"
`;

exports[`footnote_reference_strangeness.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>If <footnote_reference ids=\\"1\\" auto=\\"1\\" refname=\\"note\\"></footnote_reference> is the first footnote reference, it will show up as
&quot;[1]&quot;.  We can refer to it again as <footnote_reference ids=\\"2\\" auto=\\"1\\" refname=\\"note\\"></footnote_reference> and again see
&quot;[1]&quot;.  We can also refer to it as <reference name=\\"note\\" refname=\\"note\\">note</reference> (an ordinary internal
hyperlink reference).</paragraph><footnote ids=\\"note\\" names=\\"note\\" auto=\\"1\\"><paragraph>This is the footnote labeled &quot;note&quot;.</paragraph></footnote></document>"
`;

exports[`footnotes_multiple_refs.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document ids=\\"i-a-reference-to-footnote-1-and-i-a-reference-to\\" names=\\"\\\\ is\\\\ a\\\\ reference\\\\ to\\\\ footnote\\\\ 1,\\\\ and\\\\ 



\\\\ is\\\\ a\\\\ reference\\\\ to\\" source=\\"&lt;string&gt;\\" title=\\"

 is a reference to footnote 1, and 



 is a reference to\\">
<title><footnote_reference ids=\\"1\\" auto=\\"1\\"></footnote_reference> is a reference to footnote 1, and <footnote_reference ids=\\"2\\" auto=\\"1\\"></footnote_reference> is a reference to</title><system_message line=\\"2\\"><paragraph>Title underline too short.</paragraph><literal_block>[#]_ is a reference to footnote 1, and [#]_ is a reference to
footnote 2.</literal_block></system_message><footnote ids=\\"3\\" auto=\\"1\\"><paragraph>This is footnote 1.</paragraph></footnote><footnote ids=\\"4\\" auto=\\"1\\"><paragraph>This is footnote 2.</paragraph></footnote><footnote ids=\\"5\\" auto=\\"1\\"><paragraph>This is footnote 3.</paragraph></footnote><paragraph><footnote_reference ids=\\"6\\" auto=\\"1\\"></footnote_reference> is a reference to footnote 3.</paragraph></document>"
`;

exports[`image_directive.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<system_message line=\\"1\\">
</system_message><system_message line=\\"1\\">
<paragraph>Unknown directive type &quot;image&quot;.</paragraph><literal_block>.. image:: mylogo.png</literal_block></system_message></document>"
`;

exports[`incomplete_substitution_definition_with_image.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<comment>|beep</comment><system_message line=\\"1\\"><paragraph>malformed substitution definition.</paragraph></system_message></document>"
`;

exports[`inline_followed_by_emphasis.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph><strong>hello</strong> and <emphasis>goodbye</emphasis></paragraph></document>"
`;

exports[`inline_internal_targets.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<bullet_list bullet=\\"*\\">
<list_item>
<paragraph>bullet list</paragraph><target ids=\\"econd-item-\\" names=\\"second\\\\ item\`\\"></target></list_item><list_item><paragraph>second item, with hyperlink target.</paragraph></list_item></bullet_list></document>"
`;

exports[`interpreted_text_role_emphasis_prefix.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph><emphasis>test</emphasis></paragraph></document>"
`;

exports[`interpreted_text_role_emphasis_suffix.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph><emphasis>test</emphasis></paragraph></document>"
`;

exports[`interpreted_text_role_prefix.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph><problematic ids=\\"2\\" refid=\\"1\\">:role:\`test\`</problematic></paragraph><system_message ids=\\"1\\" backrefs=\\"2\\" line=\\"1\\"><paragraph>Unknown interpreted text role &quot;role&quot;.</paragraph></system_message></document>"
`;

exports[`interpreted_text_role_suffix.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph><problematic ids=\\"2\\" refid=\\"1\\">\`test\`:role:</problematic></paragraph><system_message ids=\\"1\\" backrefs=\\"2\\" line=\\"1\\"><paragraph>Unknown interpreted text role &quot;role&quot;.</paragraph></system_message></document>"
`;

exports[`interpreted_text_with_both_roles.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph><problematic ids=\\"2\\" refid=\\"1\\">:role:\`test\`:role:</problematic></paragraph><system_message ids=\\"1\\" backrefs=\\"2\\" line=\\"1\\"><paragraph>Multiple roles in interpreted text (both prefix and suffix present; only one allowed).</paragraph></system_message></document>"
`;

exports[`interpreted_text_with_no_end_string_suffix.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph><problematic ids=\\"2\\" refid=\\"1\\">\`</problematic>test</paragraph><system_message ids=\\"1\\" backrefs=\\"2\\" line=\\"1\\"><paragraph>Inline interpreted text or phrase reference start-string without end-string.</paragraph></system_message></document>"
`;

exports[`interpreted_text_with_no_role.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph><title_reference>test</title_reference></paragraph></document>"
`;

exports[`line_block_with_continuation_line.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<line_block>
<line indent=\\"0\\">Lend us a couple of bob till Thursday.</line><line>I&apos;m absolutely skint.</line><line>But I&apos;m expecting a postal order and I can pay you back
as soon as it comes.</line><line>Love, Ewan.</line><line indent=\\"0\\">Lend us a couple of bob till Thursday.</line><line>I&apos;m absolutely skint.</line><line>But I&apos;m expecting a postal order and I can pay you back
as soon as it comes.</line><line>Love, Ewan.</line></line_block></document>"
`;

exports[`links.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<target ids=\\"a-re-tructuredtext-primer\\" names=\\"a\\\\ restructuredtext\\\\ primer\\" refuri=\\"../../uer/rt/quicktart.html\\"></target><target ids=\\"quick-re-tructuredtext\\" names=\\"quick\\\\ restructuredtext\\" refuri=\\"../../uer/rt/quickref.html\\"></target></document>"
`;

exports[`literal_block.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<block_quote>
<paragraph>Literal blocks are either indented or line-prefix-quoted blocks,
and indicated with a double-colon (&quot;::&quot;) at the end of the
preceding paragraph (right here --&gt;):</paragraph><literal_block>if literal_block:
    text = &apos;is left as-is&apos;
    spaces_and_linebreaks = &apos;are preserved&apos;
    markup_processing = None</literal_block></block_quote></document>"
`;

exports[`literal_block_without_blank_finish.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<block_quote>
<paragraph>Literal blocks are either indented or line-prefix-quoted blocks,
and indicated with a double-colon (&quot;::&quot;) at the end of the
preceding paragraph (right here --&gt;):</paragraph><literal_block>if literal_block:
    text = &apos;is left as-is&apos;
    spaces_and_linebreaks = &apos;are preserved&apos;
    markup_processing = None</literal_block></block_quote></document>"
`;

exports[`mixed_bullets.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<bullet_list bullet=\\"*\\">
<list_item>
<paragraph>bullet</paragraph></list_item></bullet_list><system_message line=\\"2\\"><paragraph>Bullet list ends without a blank line; unexpected unindent.</paragraph></system_message><bullet_list bullet=\\"+\\"><list_item><paragraph>bullet</paragraph></list_item></bullet_list></document>"
`;

exports[`multilevel_blockquote.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>This is a top-level paragraph.</paragraph><block_quote><paragraph>This paragraph belongs to a first-level block quote.</paragraph><block_quote><paragraph>This paragraph belongs to a second-level block quote.</paragraph></block_quote></block_quote><paragraph>Another top-level paragraph.</paragraph><block_quote><block_quote><paragraph>This paragraph belongs to a second-level block quote.</paragraph></block_quote><paragraph>This paragraph belongs to a first-level block quote.  The
second-level block quote above is inside this first-level
block quote.</paragraph></block_quote></document>"
`;

exports[`nested_bullets.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<bullet_list bullet=\\"*\\">
<list_item>
<paragraph>bullet</paragraph></list_item></bullet_list><block_quote><bullet_list bullet=\\"+\\"><list_item><paragraph>bullet</paragraph></list_item><list_item><paragraph>bullet</paragraph></list_item></bullet_list></block_quote><bullet_list bullet=\\"*\\"><list_item><paragraph>bullet</paragraph></list_item></bullet_list></document>"
`;

exports[`nested_sections.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<section ids=\\"title\\" names=\\"title\\">
<title>Title</title><paragraph>Section-------</paragraph><paragraph>Third@@@@@</paragraph></section><section ids=\\"econd\\" names=\\"second\\"><title>Second</title><section ids=\\"ooop\\" names=\\"ooops\\"><title>Ooops</title></section><title>Second</title><subtitle ids=\\"ooop\\" names=\\"ooops\\">Ooops</subtitle></section></document>"
`;

exports[`option_list.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<block_quote>
<option_list>
<option_list_item>
<option_group>
<option>
<option_string>-a</option_string></option></option_group><description><paragraph>command-line option &quot;a&quot;</paragraph></description></option_list_item><option_list_item><option_group><option><option_string>-b</option_string><option_argument delimiter=\\"\\"> file</option_argument></option></option_group><description><paragraph>options can have arguments
and long descriptions</paragraph></description></option_list_item><option_list_item><option_group><option><option_string>--long</option_string></option></option_group><description><paragraph>options can be long also</paragraph></description></option_list_item><option_list_item><option_group><option><option_string>--input</option_string><option_argument delimiter=\\"=\\">file</option_argument></option></option_group><description><paragraph>long options can also have
arguments</paragraph></description></option_list_item><option_list_item><option_group><option><option_string>/V</option_string></option></option_group><description><paragraph>DOS/VMS-style options too</paragraph></description></option_list_item></option_list></block_quote></document>"
`;

exports[`paragraph_ending_in___.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>This is my paragraph ending in:</paragraph><system_message line=\\"1\\"><paragraph>Literal block expected; none found.</paragraph></system_message></document>"
`;

exports[`program_lang.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>Want to learn about <reference name=\\"my favorite programming language\\" refname=\\"my favorite programming language\\">my favorite programming language</reference>?</paragraph><target ids=\\"my-favorite-programming-language\\" names=\\"my\\\\ favorite\\\\ programming\\\\ language\\" refuri=\\"http://www.python.org\\"></target></document>"
`;

exports[`random.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<bullet_list bullet=\\"*\\">
<list_item>
<paragraph>bullet</paragraph></list_item><list_item><paragraph>bullet</paragraph></list_item></bullet_list></document>"
`;

exports[`random_2.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>Test.
Test2
Test3
-----</paragraph></document>"
`;

exports[`random_4.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document ids=\\"te-t3\\" names=\\"test3\\" source=\\"&lt;string&gt;\\" title=\\"Test3\\">
<title>Test3</title><paragraph>This is a test.</paragraph><bullet_list bullet=\\"*\\"><list_item><paragraph>BUllet list 1</paragraph></list_item><list_item><paragraph>The emacs rst editor is weird.</paragraph></list_item></bullet_list></document>"
`;

exports[`random_indent.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
</document>"
`;

exports[`short_overline.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<system_message line=\\"1\\">
<paragraph>Possible incomplete section title.
Treating the overline as ordinary text because it&apos;s so short.</paragraph></system_message><paragraph>===
Title
===</paragraph></document>"
`;

exports[`short_overline_2.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<system_message line=\\"1\\">
<paragraph>Possible incomplete section title.
Treating the overline as ordinary text because it&apos;s so short.</paragraph></system_message><paragraph>===
Title</paragraph></document>"
`;

exports[`simple_table.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<block_quote>
<table>
<tgroup cols=\\"3\\">
<colspec colwidth=\\"20\\">
</colspec><colspec colwidth=\\"10\\">
</colspec><colspec colwidth=\\"10\\">
</colspec><thead>
<row>
<entry>
<paragraph>Header row, column 1</paragraph></entry></row></thead><tbody><row><entry><paragraph>body row 1, column 1</paragraph></entry></row><row><entry><paragraph>body row 2</paragraph></entry></row></tbody></tgroup></table></block_quote></document>"
`;

exports[`simple_table_from_policies.txt.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<block_quote>
<table>
<tgroup cols=\\"3\\">
<colspec colwidth=\\"13\\">
</colspec><colspec colwidth=\\"7\\">
</colspec><colspec colwidth=\\"44\\">
</colspec><thead>
<row>
<entry>
<paragraph>Release Level</paragraph></entry></row></thead><tbody><row><entry><paragraph>alpha</paragraph></entry></row><row><entry><paragraph>beta</paragraph></entry></row><row><entry><paragraph>candidate</paragraph></entry></row><row><entry><paragraph>final</paragraph></entry></row></tbody></tgroup></table></block_quote></document>"
`;

exports[`strong.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph><strong>hello</strong></paragraph></document>"
`;

exports[`substitution_definition_with_image.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<system_message line=\\"1\\">
<paragraph>Unknown directive type &quot;image&quot;.</paragraph><literal_block>symbol here| image:: symbol.png</literal_block></system_message><system_message source=\\"\\" line=\\"1\\"><paragraph>Substitution definition &quot;symbol here&quot; empty or invalid.</paragraph><literal_block>.. |symbol here| image:: symbol.png</literal_block></system_message></document>"
`;

exports[`tabs.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>hello   tabs</paragraph></document>"
`;

exports[`title.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document ids=\\"title\\" names=\\"title\\" source=\\"&lt;string&gt;\\" title=\\"Title\\">
<title>Title</title><paragraph>Paragraph.</paragraph></document>"
`;

exports[`transition_correction.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<paragraph>====:</paragraph><literal_block>====::</literal_block></document>"
`;

exports[`transition_marker.txt 1`] = `
"<!-- generated by Docutils 0.14js -->
<document source=\\"&lt;string&gt;\\">
<transition>
</transition></document>"
`;
